aioairq/encrypt.py

```diff
@@ -22,22 +23,36 @@ def __init__(self, passw: str):
         passw : str
             Device's password
         """
-        self.key = self._pass2aes(passw)
+        self._key = self._pass2aes(passw)
 
-    def decode(self, encrypted: bytes) -> str:
+    def encode(self, data: str) -> str:
+        iv = Random.new().read(AES.block_size)
+        cipher = AES.new(self._key, AES.MODE_CBC, iv)
+
+        encoded = data.encode("utf-8")
+        encrypted = iv + cipher.encrypt(self._pad(encoded))
+
+        return base64.b64encode(encrypted).decode("utf-8")
+
+    def decode(self, encrypted: str) -> str:
         decoded = base64.b64decode(encrypted)
         iv = decoded[: self._bs]
-        cipher = AES.new(self.key, AES.MODE_CBC, iv)
+        cipher = AES.new(self._key, AES.MODE_CBC, iv)
         decrypted = cipher.decrypt(decoded[self._bs :])
         try:
             # Currently the device does not support proper authentication.
             # The success or failure of the authentication based on the ability
             # to decode the response from the device.
             decoded = decrypted.decode("utf-8")
-        except UnicodeDecodeError:
-            raise InvalidAuth("Failed to decode a message. Incorrect password")
+        except UnicodeDecodeError as e:
+            raise InvalidAuth("Failed to decode a message. Incorrect password") from e
```

Comment:
I did not know about `from None`.

I was just following the linter, did not know I could suppress with `from None`.
I agree 100% with your arguments. The decoding error is an implementation detail which should not be exposed. The user is interested in "because the password is wrong".